odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,546 @@
+
+
+class LinDeformFixedTemplDerivAdj(Operator):
+
+    """Adjoint operator of the derivative operator.
+
+    This operator computes the adjoint of the derivative of the fixed
+    template linearized deformation operator for the fixed template ``I``
+    at given displacement ``v``:
+
+        J --> grad(I)(Id + v) J
+
+    where the ``J`` is the given template and
+    the ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement mapping:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain.
+    """
+
+    def __init__(self, template, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        template : `DiscreteLp` element
+            Fixed template deformed by the vector field.
+        displacement: `ProductSpace` element
+            Point that the adjoint of the derivative needs
+            to be computed at.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> given_template = space.element([1, 2, 1, 2, 1])
+        >>> disp_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> op = LinDeformFixedTemplDerivAdj(template, disp_field)
+        >>> op(given_template)
+        ProductSpace(uniform_discr(0.0, 1.0, 5), 1).element([
+            [0.0, 10.0, -5.0, -10.0, 0.0]
+        ])
+        """
+        if displacement[0] not in template.space:
+            raise TypeError('`displacement[0]` {!r} not an element of'
+                            '`template.space`'.format(displacement[0],
+                                                      template.space))
+
+        Operator.__init__(self, template.space, displacement.space,
+                          linear=True)
+
+        self.template = template
+        self.displacement = displacement
+
+    def _call(self, func):
+        """Implement ``self(func)```.
+
+        Parameters
+        ----------
+        func : `domain` element
+            The evaluation point, i.e. an element in the template space,
+            for the deformed gradient of the template.
+        """
+        grad = Gradient(self.domain, method='forward',
+                        padding_method='symmetric')
+        template_grad = grad(self.template)
+
+        def_grad = deform_grad(template_grad, self.displacement)
+
+        return [gf * func for gf in def_grad]
+
+    @property
+    def adjoint(self):
+        """Adjoint of the ajoint operator.
+
+        Returns
+        -------
+        adjoint : `LinDeformFixedTemplDeriv`
+            The adjoint of the operator.
+        """
+        return LinDeformFixedTemplDeriv(self.template, self.displacement)
+
+
+class LinDeformFixedDisp(Operator):
+
+    """Operator using fixed displacement to map template to deformed template.
```

Comment:
This sounds weird, like half the sentence is missing. How about "Deformation operator with fixed displacement acting on templates."?